doc/BGP/BGP-route-aggregation-with-bbr-awareness.md

```diff
@@ -0,0 +1,310 @@
+Then we introduce how the bgp container will change its behavior accordingly.
+
+### Config DB Extension
+We define a new YANG model to add a new key named `BGP_AGGREGATE_ADDRESS` in config DB.
+The key will index to a list of aggregated addresses with their parameters.
+
+The YANG model and config DB demo are showed in below:
+
+#### Yang Model
+```
+module sonic-bgp-aggregate-address {
+    namespace "http://github.com/sonic-net/sonic-bgp-aggregate-address";
+
+    prefix bgp-aggregate-address;
+
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BGP aggregate address configuration module.";
+
+    revision 2024-07-17 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-aggregate-address {
+        container BGP_AGGREGATE_ADDRESS {
+
+            description "BGP_AGGREGATE_ADDRESS part of config_db.json";
+
+            list BGP_AGGREGATE_ADDRESS_LIST {
+
+                description "BGP_AGGREGATE_ADDRESS list part of config_db.json";
+
+                key "aggregate-address";
+
+                leaf aggregate-address {
+                    type inet:ip-prefix;
+                    description "Aggregate address to be advertised";
+                }
+
+                leaf bbr-required {
+                    type boolean;
+                    description "Set if BBR is required for generating aggregate address";
+                }
+
+                leaf summary-only {
+                    type boolean;
+                    description "Only advertise the summary of aggregate address";
+                }
+
+                leaf as-set {
+                     type boolean;
+                     description "Set if include the AS set when advertising the aggregated address";
+                }
+            }
+        }
+    }
+}
+```
+
+#### Config DB Sample
+```json
+{
+    ...
+    "BGP_AGGREGATE_ADDRESS": {
+        "192.168.0.0/24": {
+            "bbr-required": "true",
+            "summary-only": "false"
+        },
+        "fc00::/63": {
+            "bbr-required": "true",
+            "summary-only": "true"
+            "as-set": "true"
+        }
+    }
+    ...
+}
+```
+
+#### Parameters
+We have parameters to control the behavior of aggregate address in bgp container.
+
+##### BBR Required
+It's a boolean to indicate whether only generate aggregated address when BBR feature is enabled.
+
+If it's true and the BBR feature is not enabled, aggregated address won't generated.
+
+##### Summary Only
+It's a boolean to indicate whether only advertise aggregate address only.
+
+If it's true, then details routes will be suppressed and only aggregated address will be advertised.
+
+##### AS Set
+It's a boolean to indicate whether add a as set of details routes in as path when advertising aggregated address.
```

Comment:
Did FRR had support this feature, or we need impl it?

Reply:
FRR supports as-set, so we only need to append the parameter when execute the command.